flowexperimental/comp/flowexp_comp.cpp

```diff
@@ -44,23 +86,20 @@ int main(int argc, char** argv)
         = Opm::FlowGenericVanguard::canonicalDeckPath(Opm::Parameters::Get<Opm::Parameters::EclDeckFileName>());
 
     // Only read the RUNSPEC section of the deck
-    const auto deck = Opm::Parser{}.parseFile(inputFilename,
-                                              Opm::ParseContext{},
-                                              std::vector { Opm::Ecl::SectionType::RUNSPEC });
+    const auto deck
+        = Opm::Parser {}.parseFile(inputFilename, Opm::ParseContext {}, std::vector {Opm::Ecl::SectionType::RUNSPEC});
     const auto runspec = Opm::Runspec(deck);
     const auto numComps = runspec.numComps();
 
-    OPM_ERROR_IF(numComps < 2 || numComps > 7,
-                 fmt::format("Deck has {} components, not supported. We support a maximum of 7 components, "
-                             "and a minimum of 2.", numComps) );
-
-    switch (numComps) {
-    case 2: return Opm::dispatchFlowExpComp<2>(argc, argv);
-    case 3: return Opm::dispatchFlowExpComp<3>(argc, argv);
-    case 4: return Opm::dispatchFlowExpComp<4>(argc, argv);
-    case 5: return Opm::dispatchFlowExpComp<5>(argc, argv);
-    case 6: return Opm::dispatchFlowExpComp<6>(argc, argv);
-    case 7: return Opm::dispatchFlowExpComp<7>(argc, argv);
+    auto [componentSupported, executionStatus]
+        = runComponent<OPM_COMPILE_COMPONENTS_TEMPLATE_LIST>(numComps, argc, argv);
+
+    if (!componentSupported) {
+        fmt::print("Deck has {} components, not supported. In this build of the simulator, we support the "
+                     "following number of components:\n\n\t{}.\n\n Note that the supported components can be changed "
+                     "when configuring CMake through the OPM_COMPILE_COMPONENTS options. Exiting.\n",
```

Comment:
options-> option?

maybe it is helpful to give an example to show the full grammar here for reminding? 
```
-DOPM_COMPILE_COMPONENTS="2;4;7"
```